File: Python/programmers/etc/level2/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = []
    q = deque([idx, priority] for idx, priority in enumerate(priorities))
    
    while q:
        idx, priority = q.popleft()
        if q and priority < max([i[1] for i in q]):
            q.append([idx, priority])
        else:
            answer.append(idx)
            
    return answer.index(location)+1


priorities = [2, 1, 3, 2]
location = 2
print(solution(priorities, location)) # 1

priorities = [1, 1, 9, 1, 1, 1]
location = 0
print(solution(priorities, location)) # 5

File: Java/programmers/etc/level2/프로세스.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 프로세스 {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        PriorityQueue<Integer> q = new PriorityQueue<>(Collections.reverseOrder());

        for (int i=0; i<priorities.length; i++) {
            q.add(priorities[i]);
        }

        while (!q.isEmpty()) {
            for (int i=0; i<priorities.length; i++) {
                if (priorities[i] == q.peek()) {
                    if (i == location) {
                        answer++;
                        return answer;
                    }
                    q.poll();
                    answer++;
                }
            }
        }

        return -1;
    }
}
